### PR TITLE
Initial porting firmware test suites.

### PR DESCRIPTION
**Pull request by ftjahjad**
There are two tests:
* firmware tests library under include.
* main firmware test suites under user-specified.

### CODE REVIEW

File: firmware/include/runtest.sh [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+# Include cki library
+. ../../cki_lib/libcki.sh
+
+# Task parameters
+# DeBug - Set to non-zero value to enable debugging
+# FwtsGitRemote - git repository
+# FwtsGitBranch - git branch that will be used
+
+
+: ${DeBug:=0} # Set to non-zero value to enable debugging
```
Comment (emanchado) [inline firmware/include/runtest.sh]: I guess this is a comment more for the original maintainer than you, but probably the comment on this line should be dropped, as it's already right above.

File: firmware/include/runtest.sh [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    do
+        fwtsTest=$(echo "$line" | awk -F \| '{print $1}')
+        fwtsPass=$(echo "$line" | awk -F \| '{print $2}')
+        fwtsFail=$(echo "$line" | awk -F \| '{print $3}')
+        fwtsAbort=$(echo "$line" | awk -F \| '{print $4}')
+        fwtsWarn=$(echo "$line" | awk -F \| '{print $5}')
+        fwtsSkip=$(echo "$line" | awk -F \| '{print $6}')
+        fwtsInfo=$(echo "$line" | awk -F \| '{print $7}')
+
+        # echo "$fwtsTest $fwtsPass $fwtsFail $fwtsAbort $fwtsWarn $fwtsSkip $fwtsInfo"
```
Comment (emanchado) [inline firmware/include/runtest.sh]: Maybe drop commented-out code?

File: firmware/include/runtest.sh [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        if echo $fwtsFail | grep -q '[0-9]'; then
+            # add waive text,  when fail tests are treated pass
+            if  [[ "$FWTS_ON_FAIL_REPORT" == "PASS" ]]; then
+                rlPhaseStartTest "${fwtsTest}-[WAIVE]-see_log_files"
+                rlPass "fwtsFail count: $fwtsFail"
+                #rlPhaseEnd
+            else
+                rlPhaseStart $FWTS_ON_FAIL_REPORT $fwtsTest
+                rlFail "fwtsFail count: $fwtsFail"
+                #rlPhaseEnd
```
Comment (emanchado) [inline firmware/include/runtest.sh]: Same with these two `rlPhaseEnd`

File: firmware/user-specified/runtest.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+# Include cki library
+. ../../cki_lib/libcki.sh
+
+# Default  Firmware tests
+FWTSTESTS=${FWTSTESTS:-"--utils --batch --acpitests --acpicompliance"}
+
+rlJournalStart
+   if [[ -n $FWTSTESTS ]]; then	
+       rlPhaseStartSetup
+       fwtsSetup
```
Comment (rasibley) [inline firmware/user-specified/runtest.sh]: nitpick please indent here :-)

File: firmware/user-specified/runtest.sh [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+       rlPhaseEnd
+
+       rlPhaseStartTest
+       rlLog "Running fwts with these tests: $FWTSTESTS"
```
Comment (rasibley) [inline firmware/user-specified/runtest.sh]: Same here, and other places within the test, the code should be indented within each rlPhase, see https://github.com/beakerlib/beakerlib/wiki/man#phases-1

File: firmware/user-specified/runtest.sh [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+       
+       rlPhaseStartCleanup
+       fwtsCleanup
+       rlPhaseEnd
+   else
+       cki_abort_task "FWTSTESTS parameter is invalid, must be: --utils, --batch. --acpitests, or/and --acpicompliance"
+   fi
+rlJournalEnd
+rlJournalPrintText
+
```
Comment (rasibley) [inline firmware/user-specified/runtest.sh]: maybe remove these extra lines here

File: firmware/user-specified/Makefile [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+	@echo "Requires:	autoconf"		>> $(METADATA)
+	@echo "Requires:	automake"		>> $(METADATA)
+	@echo "Requires:	gcc"			>> $(METADATA)
+	@echo "Requires:	libtool"		>> $(METADATA)
+	@echo "Requires:	bison"			>> $(METADATA)
+	@echo "Requires:	flex"			>> $(METADATA)
+	@echo "RepoRequires:    cki_lib"                >> $(METADATA)
+	@echo "RepoRequires:    firmware/include"       >> $(METADATA)
+	@echo "License:		GPLv2"			>> $(METADATA)
+	@echo "RhtsOptions:	-Compatible -CompatService"	>> $(METADATA)
```
Comment (rasibley) [inline firmware/user-specified/Makefile]: What is this used for ? Is it still needed ?

**Comment by ftjahjad** [comment]
Remove unneeded comments, and add indentation.

**Pull request merged**